bin/varnishd/hash/hash_simple_list.c

```diff
@@ -107,18 +107,24 @@ hsl_lookup(struct worker *wrk, const void *digest, struct objhead **noh)
  * Dereference and if no references are left, free.
  */
 
-static int v_matchproto_(hash_deref_f)
-hsl_deref(struct objhead *oh)
+static int v_matchproto_(hash_deref_unlock_f)
+hsl_deref_unlock(struct worker *wrk, struct objhead *oh)
 {
 	int ret;
 
+	CHECK_OBJ_NOTNULL(oh, OBJHEAD_MAGIC);
+	Lck_AssertHeld(&oh->mtx);
+	Lck_Unlock(&oh->mtx);
+
 	Lck_Lock(&hsl_mtx);
 	if (--oh->refcnt == 0) {
```

Comment:
and same as in hash_classic

Reply:
Same reply as in comment on hash_classic.

Reply:
yes, I was wrong, thank you